mcs/class/System/Test/System.Diagnostics/PerformanceCounterTest.cs

```diff
@@ -0,0 +1,363 @@
+// This test code derives from two places:
+//
+// 1.
+// originally copied from external/corefx
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+//
+// 2. MSDN
+// https://msdn.microsoft.com/en-us/library/system.diagnostics.countercreationdata(v=vs.110).aspx
+//
+// Neither has been very significantly changed.
+//
+
+using NUnit.Framework;
+using System;
+using System.Threading;
+using System.Diagnostics;
+using System.Collections;
+using System.Collections.Specialized;
+
+namespace MonoTests.System.Diagnostics
+{
+	internal class Common
+	{
+		internal string guid;
+		internal string baseName;
+		internal string name;
+		internal string category;
+		internal CounterCreationDataCollection counterDataCollection;
+		internal CounterCreationData counter;
+		internal CounterCreationData counterBase;
+		internal PerformanceCounter counterSample;
+		internal PerformanceCounterCategory pcc;
+
+		internal void DoCommon()
+		{
+			guid = Guid.NewGuid().ToString("N");
```

Comment:
applies throughout the file: please follow the [coding guidelines](http://www.mono-project.com/community/contributing/coding-guidelines/), i.e. space before an opening parenthesis etc.